esp/esdllib/README.md

```diff
@@ -0,0 +1,1608 @@
+
+Below is a first-level outline of the common sections in the XML context:
+
+*Verify these sections, they may have changed*
+
+```xml
+<esdl_script_context>
+    <esdl/>
+    <original_request/>
+    <target/>
+    <config/>
+    <esdl_request/>
+    <final_request/>
+    <initial_response/>
+    <pre_esdl_response/>
+    <logdata/>
+    <logging/>
+    <store/>
+    <temporaries/>
+</esdl_script_context>
+```
+
+Context Node | Contents
+-------------|-------------------
+`<esdl_script_context>` | Root node, not accessible by scripts
+`<esdl>`                | ESDL interface definition info about the current method and service in progress.
+`<original_request>`    | The unmodified request as the ESP recieved from the client.
+`<target>`              | Configuration of the back end target from the ESDL service binding.
+`<config>`              | The `EspBinding` section from the ESDL service binding. This contains any defined security manager Authenticate configuration.
+`<esdl_request>`        | The result of sending the `original_request` through the esdl request transformer.
+`<final_request>`       | The result of any transformations the `BackendRequest` entry point has made on the `esdl_request` node.
+`<initial_response>`    | The unmodified response from the back end server, frequently a roxie.
+`<pre_esdl_response>`   | The result of any transformations the `BackendResponse` entry point has made on the `initial_response` node.
+`<logdata>`             | This node is passed on to the LoggingManager. The default source and target location of the `PreLogging` entry point.
+`<logging>`             | Log option and log profile settings are held here and passed on to the LoggingManager.
+`<store>`               | Location to store persistent values during the lifespan of the transaction.
+`<temporaries>`         | Location where temporaries and other variables are created by default.
+
+Some script operations will add additional elements to the context document.
+
+## Configuration
+
+An ESDL service can be configured and deployed in two ways. First is by using the configmgr web application in conjunction with ECLWatch services or the `esdl` command line tool. Second is by launching the ESP as an `esdl application`. We will look at each approach in turn.
+
+TODO: Add more information about each option. Compare them and give some basic information about deploying and running the process. Point to other documentation where appropriate.
+
+### Using configmgr
+An ESDL service is configured from two sources, one static and one dynamic. Over time we will be moving more settings into the dynamic configuration. The static configuration is part of the ESP component configuration and is loaded once when the ESP is lauched. This control settings such as
+
```

Comment:
Maybe we should stop calling it static and dynamic configurations.

I would vote for "process configuration" and "service bundle configuration".
In cloud, "process configuration" could be "pod configuration".

Reply:
updated wording as suggested